internal/config/config.go

```diff
@@ -506,8 +506,18 @@ func parseBGPAdvertisements(ads []bgpAdvertisement, cidrsPerAddresses map[string
 		}, nil
 	}
 
+	err := validateDuplicateAdvertisements(ads)
```

Comment:
Can you change the wrapping error (from https://github.com/metallb/metallb/blob/19d5b67b8276343603afc95a7f2a62258ab8eeac/internal/config/config.go#L450 ) so it mentions the addresspool it's failing? Otherwise we'll just say "community x" or "advertisement y", but if there are multiple address pools it would still be hard to find which one.

Reply:
here i can't do that because the name of the addresspool is not passed to the function. i can pass the addresspool name to the function but i don't think it's needed because the error here will be anyway wrapped by this one that's pointing to the addresspool number. 
https://github.com/metallb/metallb/blob/35bbb235d0e7f176856166dc16c997a09367b62b/internal/config/config.go#L310

Reply:
Ah cool